File: TelerikUIExamples/TelerikUIExamples/ChartDocsAreaSeries.m
//
//  ChartDocsAreaSeries.m
//  TelerikUIExamples
//
//  Copyright © 2016 Telerik. All rights reserved.
//

#import "ChartDocsAreaSeries.h"
#import <TelerikUI/TelerikUI.h>

@implementation ChartDocsAreaSeries
{
    NSMutableArray *pointsWithCategoriesAndValues;
    NSMutableArray *pointsWithCategoriesAndValues2;
    TKChart *chart;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // >> chart-area
    pointsWithCategoriesAndValues = [[NSMutableArray alloc] init];
    pointsWithCategoriesAndValues2 = [[NSMutableArray alloc] init];
    NSArray *categories = @[ @"Greetings", @"Perfecto", @"NearBy", @"Family Store", @"Fresh & Green" ];
    NSArray *values = @[ @70, @75, @58, @59, @88 ];
    for (int i = 0; i < categories.count; i++) {
        TKChartDataPoint *dataPoint = [[TKChartDataPoint alloc] initWithX:categories[i] Y:values[i]];
        [pointsWithCategoriesAndValues addObject:dataPoint];
    }
    
    NSArray *values2 = @[ @40, @80, @35, @69, @95 ];
    for (int i = 0; i < categories.count; i++) {
        TKChartDataPoint *dataPoint = [[TKChartDataPoint alloc] initWithX:categories[i] Y:values2[i]];
        [pointsWithCategoriesAndValues2 addObject:dataPoint];
    }
    
    TKChartAreaSeries* seriesForIncomes = [[TKChartAreaSeries alloc] initWithItems:pointsWithCategoriesAndValues];
    [chart addSeries:seriesForIncomes];
    
    TKChartAreaSeries *seriesForExpenses = [[TKChartAreaSeries alloc] initWithItems:pointsWithCategoriesAndValues2];
    [chart addSeries:seriesForExpenses];
    // << chart-area
    
    [self visualAppearance];
}

- (void) visualAppearance
{
    // >> chart-style-fill
    TKChartAreaSeries* seriesForAnnualRevenue = [[TKChartAreaSeries alloc] initWithItems:pointsWithCategoriesAndValues];
    seriesForAnnualRevenue.style.palette = [[TKChartPalette alloc] init];
    TKChartPaletteItem *palleteItem = [[TKChartPaletteItem alloc] init];
    palleteItem.stroke = [TKStroke strokeWithColor:[UIColor brownColor]];
    palleteItem.fill = [TKSolidFill solidFillWithColor:[UIColor redColor]];
    [seriesForAnnualRevenue.style.palette addPaletteItem:palleteItem];
    [chart addSeries:seriesForAnnualRevenue];
    // << chart-style-fill
}

- (void) stackingAreaSeries
{
    // >> chart-stack-area
    TKChartStackInfo *stackInfo = [[TKChartStackInfo alloc] initWithID:@(1) withStackMode:TKChartStackModeStack];
    
    TKChartAreaSeries *seriesForIncome = [[TKChartAreaSeries alloc] initWithItems:pointsWithCategoriesAndValues];
    seriesForIncome.stackInfo = stackInfo;
    
    TKChartAreaSeries *seriesForExpences = [[TKChartAreaSeries alloc] initWithItems:pointsWithCategoriesAndValues2];
    seriesForExpences.stackInfo = stackInfo;
    
    [chart beginUpdates];
    [chart addSeries:seriesForIncome];
    [chart addSeries:seriesForExpences];
    [chart endUpdates];
    // << chart-stack-area
}

- (void) stacking100AreaSeries
{
    // >> chart-stack-area-100
    TKChartStackInfo *stackInfo = [[TKChartStackInfo alloc] initWithID:@(1) withStackMode:TKChartStackModeStack100];
    
    TKChartAreaSeries *seriesForIncome = [[TKChartAreaSeries alloc] initWithItems:pointsWithCategoriesAndValues];
    seriesForIncome.stackInfo = stackInfo;
    
    TKChartAreaSeries *seriesForExpences = [[TKChartAreaSeries alloc] initWithItems:pointsWithCategoriesAndValues2];
    seriesForExpences.stackInfo = stackInfo;
    
    [chart beginUpdates];
    [chart addSeries:seriesForIncome];
    [chart addSeries:seriesForExpences];
    [chart endUpdates];
    // << chart-stack-area-100
}

- (void) areaSeries
{
    TKChartAreaSeries* seriesForIncomes = [[TKChartAreaSeries alloc] initWithItems:pointsWithCategoriesAndValues];
    [chart addSeries:seriesForIncomes];
    
    TKChartAreaSeries *seriesForExpenses = [[TKChartAreaSeries alloc] initWithItems:pointsWithCategoriesAndValues2];
    [chart addSeries:seriesForExpenses];
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/ChartDocsAreaSeries.swift
//
//  ChartDocsAreaSeries.swift
//  TelerikUIExamplesInSwift
//
//  Copyright © 2016 Telerik. All rights reserved.
//

class ChartDocsAreaSeries: UIViewController {
    var pointsWithCategoriesAndValues = [TKChartDataPoint]()
    var pointsWithCategoriesAndValues2 = [TKChartDataPoint]()
    var chart:TKChart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        chart.frame = CGRectInset(self.view.bounds, 10, 10)
        self.view.addSubview(chart)
        // >> chart-area-swift
        let categories = ["Greetings", "Perfecto", "NearBy", "Family Store", "Fresh & Green" ];
        let values = [70, 75, 58, 59, 88]
        for i in 0 ..< categories.count {
            pointsWithCategoriesAndValues.append(TKChartDataPoint(x: categories[i], y: values[i]))
        }
        
        let values2 = [40, 80, 32, 69, 95]
        for i in 0 ..< categories.count {
            pointsWithCategoriesAndValues2.append(TKChartDataPoint(x: categories[i], y: values2[i]))
        }
        
        chart.addSeries(TKChartAreaSeries(items: pointsWithCategoriesAndValues))
        chart.addSeries(TKChartAreaSeries(items: pointsWithCategoriesAndValues2))
        // << chart-area-swift
    }
    
    func areaSeries() {
        let seriesForIncome = TKChartAreaSeries(items: pointsWithCategoriesAndValues)
        chart.addSeries(seriesForIncome)
        
        let seriesForExpenses = TKChartAreaSeries(items: pointsWithCategoriesAndValues2)
        chart.addSeries(seriesForExpenses)
    }
    
    func stacking100AreaSeries() {
        // >> chart-stack-area-100-swift
        let stackInfo = TKChartStackInfo(ID: 1, withStackMode: TKChartStackMode.Stack100)
        
        let seriesForIncome = TKChartAreaSeries(items: pointsWithCategoriesAndValues)
        seriesForIncome.stackInfo = stackInfo
        
        let seriesForExpenses = TKChartAreaSeries(items: pointsWithCategoriesAndValues2)
        seriesForExpenses.stackInfo = stackInfo
        
        chart.beginUpdates()
        chart.addSeries(seriesForIncome)
        chart.addSeries(seriesForExpenses)
        chart.endUpdates()
        // << chart-stack-area-100-swift
    }
    
    func stackingAreaSeries() {
        // >> chart-stack-area-swift
        let stackInfo = TKChartStackInfo(ID: 1, withStackMode: TKChartStackMode.Stack)
        
        let seriesForIncome = TKChartAreaSeries(items: pointsWithCategoriesAndValues)
        seriesForIncome.stackInfo = stackInfo
        
        let seriesForExpenses = TKChartAreaSeries(items: pointsWithCategoriesAndValues2)
        seriesForExpenses.stackInfo = stackInfo
        
        chart.beginUpdates()
        chart.addSeries(seriesForIncome)
        chart.addSeries(seriesForExpenses)
        chart.endUpdates()
        // << chart-stack-area-swift
    }
    
    func visualAppearance() {
        // >> chart-style-fill-swift
        let seriesForAnnualRevenue = TKChartAreaSeries(items: pointsWithCategoriesAndValues)
        seriesForAnnualRevenue.style.palette = TKChartPalette()
        let paletteItem = TKChartPaletteItem()
        paletteItem.stroke = TKStroke(color: UIColor.brownColor())
        paletteItem.fill = TKSolidFill(color: UIColor.redColor())
        seriesForAnnualRevenue.style.palette!.addPaletteItem(paletteItem)
        chart.addSeries(seriesForAnnualRevenue)
        // << chart-style-fill-swift
    }
    
}
